model/provisioner_param.go

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+//
+
+package model
+
+// ProvisioningParams represent configuration used during various provisioning operations.
+type ProvisioningParams struct {
```

Comment:
nit: I can see why you moved this config object to `model`, but it may still be better suited staying in `internal`. My general guideline for whether something should be in `model` or not is if there is a situation where an outside service may want to reference or use the object. In this case, ProvisioningParams is only really used in our internal supervisor/provisioner code so not sure it would be useful.

As I said, this is just a general guideline and not something I feel strongly on. Just figured I would mention it.

Reply:
That is right. To solve the cycle dependency, I must package it separately. What about a model `package` inside the `/internal/provisioner/model` directory?

Reply:
I think I have found a solution